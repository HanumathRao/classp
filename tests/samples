class A {
  %parseable;
  int n;
  syntax('+' n);
  sample('+ 3');
  sample('+ 3', succeed);
  sample('+ 3', fail);
  sample('+ 3', print);
  sample('+ 3', '(A n:3)');
  sample('+ 3', 'testing');
  sample('- 3');
  sample('- 3', succeed);
  sample('- 3', fail);
  sample('- 3', print);
  sample('- 3', 'testing');
}
