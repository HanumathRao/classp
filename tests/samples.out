parsing '+ 3':
SUCCEEDS
parsing '+ 3':
SUCCEEDS
parsing '+ 3':
SUCCEEDS: ERROR[expected fail]
parsing '+ 3':
SUCCEEDS: (A n:3)
parsing '+ 3':
SUCCEEDS
parsing '+ 3':
SUCCEEDS: ERROR[no match:
  expected-> testing
  actual->   (A n:3)
  ]
parsing '- 3':
-FAILS: ERROR[expected success]
  ERROR: 1.1-3:syntax error, unexpected integer literal, expecting `+`
1 errors.
parsing '- 3':
-FAILS: ERROR[expected success]
  ERROR: 1.1-3:syntax error, unexpected integer literal, expecting `+`
1 errors.
parsing '- 3':
-FAILS [as expected]
parsing '- 3':
-FAILS: ERROR[expected success]
  ERROR: 1.1-3:syntax error, unexpected integer literal, expecting `+`
1 errors.
parsing '- 3':
-FAILS: ERROR[expected testing]
  ERROR: 1.1-3:syntax error, unexpected integer literal, expecting `+`
1 errors.
Errors: 6
